### DISCUSSION TITLE
Pybricks won't pair with Lego BOOST hub

### DISCUSSION THREAD

**Discussion by Boomer133** [pybricks/support#1693] [Q&A]
I have a new PC without built-in Bluetooth but have fitted Ugreen CM591. Fresh install of Chrome. I can successfully pair a Lego BOOST hub that has just been updated via Lego Powered Up App. using Scratch with Scratch Link and Chrome. I can control this hub using Scratch. 
I cannot install Pybricks firmware as 'no compatible device is found' despite repeated scans. I have selected the hub option and I have held down the hub button until it flashes purple. Using an Android tablet with Bluetooth Sniffer I can see that both the CM591 and Lego hub are active. Any ideas? Thanks.

**Comment by dlech** [comment]
You could try flashing the firmware using the Android tablet.

If you want to share the Bluetooth sniffer capture, that could provide some insights into the problem.

**Comment by Boomer133** [comment]
Move Hub trying to establish pairing with Lego App

06/30/2024 09:46:54 AM        <->34ms       -56dB        No Deserializer    Move Hub          x87      00:16:53:AE:AD:9D

0x09 0xFF 0x97 0x03 0x00 0x40 0x06 0xFE 0x43 0x00 0x02 0x01 0x06 0x11 0x07 0x23 0xD1 0xBC 0xEA 0x5F 0x78 0x23 0x16
0xDE 0xEF 0x12 0x12 0x23 0x16 0x00 0x00 0x05 0x12 0x10 0x00 0x20 0x00 0x02 0x0A 0x00 0x09 0x09 0x4D 0x6F 0x76 0x65
0x20 0x48 0x75 0x62

Move Hub flashing purple

06/30/2024 10:02:56 AM        <->70ms       -69dB        No Deserializer    LEGO Bootloader   x36      00:16:53:AE:AD:9E

0x02 0x01 0x06 0x11 0x07 0x23 0xD1 0xBC 0xEA 0x5F 0x78 0x23 0x16 0xDE 0xEF 0x12 0x12 0x25 0x16 0x00 0x00 0x09 0xFF
0x97 0x03 0x00 0x00 0x00 0x10 0x40 0x02 0x05 0x12 0x10 0x00 0x20 0x00 0x02 0x0A 0x00 0x10 0x09 0x4C 0x45 0x47 0x4F
0x20 0x42 0x6F 0x6F 0x74 0x6C 0x6F 0x61 0x64 0x65 0x72

Pybricks attempting link for firmware install

06/30/2024 10:23:21 AM        <->120ms       -41dB        No Deserializer    Unknown device   x9      2F:12:18:8E:E8:5C

0x1E 0xFF 0x06 0x00 0x01 0x09 0x20 0x22 0x71 0x9F 0xD6 0xF2 0x6D 0x83 0x42 0x3A 0xAD 0xD7 0xFD 0x85 0x3A 0xEB 0x6E
0x6F 0xBC 0xEE 0xA2 0x31 0xC3 0x64 0x2E


The sniffer is picking up other devices too but note that the same hardware has no trouble running Scratch/Scratch Link in the same environment. If I was to load firmware via Android tablet (don't have Pybricks on a tablet) not sure that pair would then work on PC and might end up with locked out hub?
Thank you for responding.

    On Sunday, 30 June 2024 at 00:13:10 BST, David Lechner ***@***.***> wrote:   

 


You could try flashing the firmware using the Android tablet.

If you want to share the Bluetooth sniffer capture, that could provide some insights into the problem.

**Reply by laurensvalk** [reply]
Just to double check --- you're accessing the firmware installation as below, right? (That is, not via the main Bluetooth button)

![image](https://github.com/pybricks/support/assets/12326241/79db8e42-b362-4745-a32d-fdfeb82a90a7)

> If I was to load firmware via Android tablet (don't have Pybricks on a tablet) not sure that pair would then work on PC and might end up with locked out hub?

You can use any Android phone/tablet to restore the firmware using the official LEGO Powered Up app. You can always try this first, just to be sure. See the following for instructions:

![image](https://github.com/pybricks/support/assets/12326241/ba93420f-95d7-4461-8785-1c9f93cb23f3)

**Comment by Boomer133** [comment]
Yes, then I make sure that the hub (left hand icon) is selected, follow the instruction to hold down the hub button and keep it down while the hub is flashing purple. Still holding the button down I click on Install and the scanning box comes up and ultimately times out, at which point I release the hub button and wonder what to try next.
I wouldn't want to use Pybricks on a tablet for coding, if I can't get it to work on the PC I will have to stick to BOOST/Power Up, which will be a shame.

    On Sunday, 30 June 2024 at 12:24:42 BST, laurensvalk ***@***.***> wrote:   

 


Just to double check --- you're accessing the firmware installation as below, right? (That is, not via the main Bluetooth button)

image.png (view on web)


If I was to load firmware via Android tablet (don't have Pybricks on a tablet) not sure that pair would then work on PC and might end up with locked out hub?


You can use any Android phone/tablet to restore the firmware using the official LEGO Powered Up app. You can always try this first, just to be sure. See the following for instructions:

image.png (view on web)

**Reply by dlech** [reply]
Just to be clear, you are using this dialog to install the firmware?

![image](https://github.com/pybricks/support/assets/963645/d22724ad-2bea-4827-af73-8b65afe4bd47)

![image](https://github.com/pybricks/support/assets/963645/432902ee-f687-4912-b03e-a850cea6ad65)

**Comment by Boomer133** [comment]
Absolutely, the hub as per the image is selected.

    On Sunday, 30 June 2024 at 23:29:43 BST, David Lechner ***@***.***> wrote:   

 


Just to be clear, you are using this dialog to install the firmware?

image.png (view on web)

image.png (view on web)

**Reply by dlech** [reply]
If you are lucky, it may be possible to flash the firmware using the Android tablet and then code using the PC.

**Comment by Boomer133** [comment]
I have now tried using Chrome on Android tablet and have managed to load the Pybricks firmware to the test hub. Everything looked as I remember it using Chrome on the PC. I suspect that I might have done one thing that was different.
When the scanning dialogue box popped up the BOOST hub is listed at the top of the box and the scanning message at the bottom. On the PC I watched that dialogue time out, on the tablet I think I clicked on the hub identity within that dialogue box which then brought up the option to pair. In other words I thought that the request to pair with the hub had already been made, causing the scanning box to pop up, whereas in reality the box was needing confirmation of the request which it wasn't getting from me.
I think the instructions could be clearer about the action needed to be taken.
I presented the hub to the PC and again scanning appeared to be taking place without pairing. I then clicked on the hub identity at the top of the dialogue box and was able to pair and subsequently run the sample speed dial code.
Thank you for your support.

    On Sunday, 30 June 2024 at 23:46:56 BST, David Lechner ***@***.***> wrote:   

 


If you are lucky, it may be possible to flash the firmware using the Android tablet and then code using the PC.

**Reply by laurensvalk** [reply]
Glad to hear it worked! Coincidentally, we're working on an improved starter guide as we speak.

It sounds like you mean the following (only then the equivalent step for the firmware installation, not pictured here).

![image](https://github.com/pybricks/support/assets/12326241/0d7f6e34-4d6d-4752-a430-fed4c126c5a6)

The dialog is something that Chrome provides, so we can't change how it looks, but we can include things like the above in the guide.

**Comment by Boomer133** [comment]
Yes! 
So obvious once you know but if you haven't done it before it just seems like confirmation of what has already been selected, especially as it appears to be actively scanning rather than waiting for user action.
I appreciate your response.

    On Monday, 1 July 2024 at 19:07:38 BST, laurensvalk ***@***.***> wrote:   

 


Glad to hear it worked! Coincidentally, we're working on an improved starter guide as we speak.

It sounds like you mean the following (only then the equivalent step for the firmware installation, not pictured here).

image.png (view on web)

The dialog is something that Chrome provides, so we can't change how it looks, but we can include things like the above in the guide.
